network/endpoint_windows.go

```diff
@@ -502,7 +502,7 @@ func (nw *network) deleteEndpointImplHnsV2(ep *endpoint) error {
 
 	// Remove this endpoint from the namespace
 	if err = hnsv2.RemoveNamespaceEndpoint(hcnEndpoint.HostComputeNamespace, hcnEndpoint.Id); err != nil {
-		return fmt.Errorf("Failed to remove hcn endpoint: %s from namespace: %s due to error: %v", ep.HnsId,
+		log.Errorf("Failed to remove hcn endpoint: %s from namespace: %s due to error: %v", ep.HnsId,
 			hcnEndpoint.HostComputeNamespace, err)
 	}
```

Comment:
Ideally we should be able to getNameSpace to check if its removed before calling this method. But not sure if HNS has a getNameSpace API to see if its created.

Reply:
as we discussed offline, even if this api return other errors, cni should still go ahead and delete hns endpoint and release ip from ipam